lib/kernel/src/trace.erl

```diff
@@ -0,0 +1,1229 @@
+%%
+%% %CopyrightBegin%
+%%
+%% Copyright Ericsson AB 2024. All Rights Reserved.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%
+%% %CopyrightEnd%
+%%
+
+-module(trace).
+-moduledoc #{since => "OTP 27.0"}.
+-moduledoc """
+The new trace interface.
```

Comment:
```suggestion
The Erlang trace interface.
```

new gets old very quickly...

Should we add some general description about what tracing is and link to `dbg`, `ttb` and such?